src/pages/DevicesensorSettings.tsx

```diff
@@ -0,0 +1,512 @@
+import { ArrowForward } from "@mui/icons-material";
+import DiscreteSliderMarks from "../components/ui/DiscreteMarks";
+import { useCallback, useContext, useEffect, useState } from "react";
+import { Actuator, Device, Sensor } from "waziup";
+import ontologies from "../assets/ontologies.json";
+import { ActuatorX, DevicesContext, SensorX } from "../context/devices.context";
+import PrimaryIconButton from "../components/shared/PrimaryIconButton";
+import React from "react";
+import OntologyKindInput from "../components/shared/OntologyKindInput";
+import { cleanString } from "../utils";
+import SnackbarComponent from "../components/shared/Snackbar";
+export interface HTMLSelectPropsString extends React.SelectHTMLAttributes<HTMLSelectElement> {
+    handleChange: (e:SelectChangeEvent<string>)=>void,
+    title: string,
+    conditions: string[],
+    value: string
+    my?: number
+    isDisabled?: boolean
+    matches?: boolean
+}
+export const SelEl = ({ handleChange, title, my,name, conditions, isDisabled, value }: HTMLSelectPropsString) => (
+    <Box minWidth={120}  my={my !== undefined ? my : 2} >
+        <Typography fontSize={12} color={DEFAULT_COLORS.secondary_black}>{title}</Typography>
+        <FormControl variant="standard" color="primary" disabled={isDisabled} fullWidth>
+            <Select
+                inputProps={{
+                    name: name,
+                    id: 'uncontrolled-native',
+                }}
+                name={name}
+                required
+                value={value}
+                onChange={handleChange}
+            >
+                {
+                    conditions.map((op,idx)=>(
+                        <MenuItem key={idx} value={op} sx={{display:'flex',width:'100%', justifyContent:'space-between'}}>
+                            <Box display={'flex'} alignItems={'center'}>
+                                <Typography fontSize={14} color={'#325460'} >{op}</Typography>
+                                
+                            </Box>
+                            
+                        </MenuItem>
+                    ))
+                }
+            </Select>
+        </FormControl>
+    </Box>
+);
+function DeviceSensorSettings() {
+    const [matches] = useOutletContext<[matches: boolean]>();
+    const { pathname } = useLocation();
+    const { id, sensorId } = useParams();
+    const [device, setDevice] = useState<Device | null>(null);
+    const [sensOrActuator, setSensOrActuator] = useState<SensorX | ActuatorX | null>(null);
+    const [rActuator, setRemoteActuator] = useState<ActuatorX | SensorX | null>(null);
+    const [error, setError] = useState<{message: Error | null | string,severity: "error" | "warning" | "info" | "success"} | null>(null);
+    const { getDevicesFc,showDialog } = useContext(DevicesContext);
+    const handleToggleEnableSwitch = () => {
+        setSensOrActuator({
+            ...sensOrActuator!,
```

Comment:
I think Sensors and Actuators are very different, so I don't feel confortable having them in the same file. It's better to create separate files for sensors and actuators (for the sensor/actuator pages and their respectives settings pages).

Reply:
I put the same file for both of them because they have some similarites.

Reply:
They will be more different as we go. It's better to separate them. It will be cleaner.